### PR TITLE
Add tests for dict_to_list utility function

### PR DESCRIPTION
**Pull request by sanjevShakya**
### Description
- Updated unit test for method dict_to_list util method

Screenshot from 2020-02-13 10-15-59

### CODE REVIEW

File: tests/util/test_object.py [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+    }
+
+    expected = [
+        {'name': 'foo', 'value': 'bar'},
+        {'name': 'just', 'value': 'test'},
+        {'name': 'hello', 'value': 'world'}
+    ]
+
+    assert dict_to_list(value) == expected
+
```
Comment (kabirbaidhya) [inline tests/util/test_object.py]: PEP8 requires two spaces in between function declarations. 

```suggestion


```

File: tests/util/test_object.py [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
+def test_dict_to_list_2():
+    '''
+    Test an empty dictionary converted to empty list.
+    '''
+    value = {}
+
+    expected = []
+
+    assert dict_to_list(value) == expected
+
```
Comment (kabirbaidhya) [inline tests/util/test_object.py]: Same here - two blank lines
```suggestion


```

File: tests/util/test_object.py [hunk 3]
```diff
@@ -1,7 +1,47 @@
...
+def test_dict_to_list_3():
+    '''
+    Test an invalid argument such as int to dict_to_list
+    '''
+    value = 1
+
+    with pytest.raises(AttributeError) as ex:
+        dict_to_list(value)
+
+    assert ex.value.args[0] == "'int' object has no attribute 'items'"
```
Comment (kabirbaidhya) [inline tests/util/test_object.py]: Good that you have this test that catches error. 
I think we'll need to catch this type error inside and throw our own specific error from the function if the type is not a dictionary. Here's what it should say.

```py
    assert ex.value.args[0] == "Argument must be a dictionary, invalid argument received '1'."
```

File: tests/util/test_object.py (hunk 2 above)
Comment (sanjevShakya) [reply tests/util/test_object.py]: Updated.

File: tests/util/test_object.py (hunk 1 above)
Comment (sanjevShakya) [reply tests/util/test_object.py]: Updated

Comment (shakyapranin) [reply tests/util/test_object.py]: Will BLACK be doing this later? @kabirbaidhya

File: tests/util/test_object.py [hunk 4]
```diff
@@ -1,7 +1,49 @@
 ''' Tests for datamegh.util.object util. '''
 
 import pytest
-from datamegh.util.object import linearize, delinearize, merge
+from datamegh.util.object import linearize, delinearize, merge, dict_to_list
+
+
+def test_dict_to_list_1():
```
Comment (shakyapranin) [inline tests/util/test_object.py]: Not sure if naming tests with post fix `_number` is a good idea.

File: datamegh/util/types.py [hunk 5]
```diff
@@ -24,6 +24,11 @@ def is_dict(obj):
     return type(obj) == type({})
 
 
+def is_not_dict(obj):
+    ''' Check if the object is not a dict. '''
+    return not is_dict(obj)
```
Comment (kabirbaidhya) [inline datamegh/util/types.py]: No need this.

File: datamegh/util/object.py [hunk 6]
```diff
@@ -5,13 +5,16 @@
...
 
-from datamegh.util.types import is_dict, is_iterable, is_string
+from datamegh.util.types import is_dict, is_not_dict, is_iterable, is_string
 
 
 def dict_to_list(dict):
     '''
     Returns a list of dictionaries with `name` and `value` keys for all key-value pair in given dictionary.
     '''
+    if is_not_dict(dict):
```
Comment (kabirbaidhya) [inline datamegh/util/object.py]: Be pythonic. 
```suggestion
    if not is_dict(dict):
```

File: tests/util/test_object.py (hunk 1 above)
Comment (kabirbaidhya) [reply tests/util/test_object.py]: Yes but black follows PEP8 too.

File: tests/util/test_types.py [hunk 7]
```diff
@@ -2,6 +2,7 @@
 
 from datamegh.util.types import (
     is_iterable,
+    is_not_dict,
```
Comment (kabirbaidhya) [inline tests/util/test_types.py]: No need. Because we have `not` operator - let it do the job.

File: tests/util/test_types.py [hunk 8]
```diff
@@ -90,3 +91,19 @@ def test_is_list_returns_false_for_arguments_other_than_a_list():
...
+    assert is_not_dict(dict()) is False
+
+
+def test_is_not_dict_returns_true_for_arguments_other_than_a_dict():
+    ''' Test is_not_dict() for valid arguments of other than dict data type. '''
+    assert is_not_dict('') is True
+    assert is_not_dict('Test') is True
+    assert is_not_dict(u'Hello') is True
+    assert is_not_dict(None) is True
+    assert is_not_dict([]) is True
```
Comment (kabirbaidhya) [inline tests/util/test_types.py]: Nice try but no need :smile:.

File: tests/util/test_object.py (hunk 4 above)
Comment (kabirbaidhya) [reply tests/util/test_object.py]: Yes, you can make the function name a bit verbose - which will do the job for eg: `test_dict_to_list_converts_dict_to_list`

**Pull request merged**